stan/math/rev/core/vari.hpp

```diff
@@ -30,7 +30,6 @@ class vari_base {
    */
   virtual void chain() = 0;
   virtual void set_zero_adjoint() = 0;
-  virtual ~vari_base() noexcept {}
```

Comment:
Also a previous goof I'm fixing in this PR. A virtual destructor allows the destructor of inherited classes to be called when an object of an inherited class is assigned to an object with the type of the base class. But the `vari_base` class never has it's destructor called since it only ever sits in our stack allocator. So this just added a pointer to a vtable we didn't need